### ISSUE TITLE
Clickable items should have OnClick callbacks

### ISSUE THREAD

**Issue by mikeburke106** [smartdevicelink/sdl_java_suite#15]
Items that can be touched or clicked on the head-unit should have associated OnClick callbacks so that the click event can be triggered by the library instead of handled by each individual developer.

For example, the [Livio SDL Tester](http://git.projects.genivi.org/?p=smartdevicelink_tester_android.git;a=blob_plain;f=LivioSdlUtilities/src/com/livio/sdl/SdlService.java;hb=3f7503cc0440fc3aceb770606e0b8a3d6d7a031f) application on GENIVI has defined a MenuManager class to handle click events from onPerformInteractionResponse, onOnButtonPress & onOnCommand.  Within these callbacks, a "disptachClick" method is called, which, in turn, calls the onClickListener that was defined for each clickable item:

``` java
@Override
public void onOnCommand(final OnCommand notification){
        int buttonId = notification.getCmdID();
        menuManager.dispatchClick(buttonId);
}
```

**Comment by Heath-FoMoCo** [comment]
I think this can be closed .  IT should be handled by then enhanced framework.
https://github.com/smartdevicelink/sdl_android/pull/258/

**Comment by joeygrover** [comment]
This will be closed once #258 is pulled into develop.

**Comment by joeljfischer** [comment]
Closed due to manager frameworks
